src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -293,4 +299,24 @@ private synchronized void resubscribe() {
         // Resubscribe to topics
         updateSubscriptionsInternal();
     }
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private synchronized void subscribeToTopics(Set<String> topics) {
```

Comment:
this shouldn't be synchronized on `this`, you're blocking too much.

Reply:
removed unnecessary `synchronized` from methods.